geode-core/src/main/java/org/apache/geode/internal/cache/BucketAdvisor.java

```diff
@@ -2766,23 +2766,23 @@ public boolean isShadowBucketDestroyed(String shadowBucketPath) {
 
   boolean checkIfAllColocatedChildBucketsBecomePrimary() {
     List<PartitionedRegion> colocatedChildPRs = getColocateNonShadowChildRegions();
-    if (colocatedChildPRs.size() > 0) {
-      for (PartitionedRegion partitionedRegion : colocatedChildPRs) {
-        Bucket bucket = partitionedRegion.getRegionAdvisor().getBucket(getBucket().getId());
-        if (bucket != null) {
-          BucketAdvisor bucketAdvisor = bucket.getBucketAdvisor();
-          if (!bucketAdvisor.checkIfAllColocatedChildBucketsBecomePrimary()) {
+    if (colocatedChildPRs.isEmpty()) {
+      return checkIfBecomesPrimary();
```

Comment:
This is intended for the leaf colocated child bucket (we need to wait for it to become primary as well). It will fail the test case if you simply change it to return true. (This is to ensure all child buckets to become primary)

Reply:
Ah, I see, thanks for the explanation